### DISCUSSION TITLE
内存占用问题

### DISCUSSION THREAD

**Discussion by arror** [xjasonlyu/tun2socks#117] [Q&A]
关闭

**Comment by xjasonlyu** [comment]
解决了？

**Reply by arror** [reply]
> 解决了？

没有，我是在苹果设备上使用，包括iOS和macOS。由于iOS的NetworkExtension内存限制，即使我设置很小的buffer size，当有大量连接时仍旧大概率会超出限制。

**Reply by xjasonlyu** [reply]
在iOS上是有这样的问题，一个runtime就有很大的占用，所以调整buffer size这种也只是适合内存有256M以上的路由器而已。

需要在iOS上使用这类功能，其实可以考虑一下复用system stack的工具，例如[tun2socket](https://github.com/Kr328/tun2socket)这个项目，或者干脆别用Go。。😂

**Reply by arror** [reply]
> 在iOS上是有这样的问题，一个runtime就有很大的占用，所以调整buffer size这种也只是适合内存有256M以上的路由器而已。
> 
> 需要在iOS上使用这类功能，其实可以考虑一下复用system stack的工具，例如[tun2socket](https://github.com/Kr328/tun2socket)这个项目，或者干脆别用Go。。😂

嗯，因为在做clash的客户端程序，之前用的是go-tun2socks，go-tun2socks不再维护，所以寻找新的替代方案。
